analyses/cell-type-wilms-tumor-06/notebook-template/01-clustering.Rmd

```diff
@@ -0,0 +1,768 @@
+
+
+## Sample metadata
+
+```{r paths2}
+## to be modified when rendered
+filedir <- filelist[i] 
+sample <- gsub(paste0(data_dir, "/"), "", filedir)
+
+metadata <- read.table(file.path(data_dir, "/single_cell_metadata.tsv"), sep = "\t", header=TRUE)
+
+metadata[metadata$scpca_sample_id == sample,]
+```
+
+
+### Description of the analysis workflow
+
+We perform the following analysis to assess for the quality of clustering:
+
+[1] We perform some quality check to assess any QC-induced clustering (nFeature, nCount, percent.mito). 
+
+[2] We add cell cycle information, as we know that in a specific cell cycle state, the transcriptional program is mostly/exclusively related to cell cycle genes and the identity of cells is difficult to determine. 
+We expect these cells to cluster together in a cluster of proliferating cells. 
+
+[3] We look at specific marker genes that we reported in the table marker.sets/CellType_metadata.csv to check the relevance of the clustering.
+
+[4] We look at specific pathways that we reported in the table marker.sets/Pathways_metadata.csv to check the relevance of the clustering.
+
+[5] We run DElegate::FindAllMarkers2 to find markers of the different clusters and manually check if they do make sense. 
+DElegate::FindAllMarkers2 is an improved version of Seurat::FindAllMarkers based on pseudobulk differential expression method. 
+
+[6] We perform enrichment analysis of marker genes for each seurat clusters. 
+We defined all the genes from the seurat object as the universe and used the MSigDB gene sets. 
+
+[7] We plot pca/umap reduction grouping with available annotations (singler_, cellassign_). 
+We expect at least immune cells to be correctly label and fall into a few set of clusters.
+
+[ ] Next step, aim for a next PR: We will label transfer (Azimuth) to transfer annotation from the fetal kidney atlas human reference. 
+We plot pca/umap reduction grouping with latest labels. 
+We expect it to be the most representative of the cell types in the sample. 
+
+
+### Selected parameters
+
+Please note: to keep the notebook as straight as possible, we decided to show the analysis for the selected set of parameters:
+
+- SCTransform (default parameters)
+- RunPCA (default) 
+- RunUMAP (dims = 1:50) 
+- FindNeighbors (dims = 1:50) 
+- FindClusters (default)
+
+
+Note: Other parameters have been previously tested, but we would like to show in the following report that the one selected is performing good.
+
+
+## Analysis content
+
+### Load the processed SingleCellExperiment rds object 
+
+
+```{r load}
+file <- list.files(filedir)
+file <- file[grepl("_processed.rds", file)]
+
+# open the processed rds object
+sce <- readRDS(paste0(filedir, "/", file))
+```
+
+
+### Run Seurat pipeline
+
+```{r seurat_wf, fig.width=8, fig.height=4, out.width='100%'}
+# convert to seurat
+srat <- CreateSeuratObject(counts = counts(sce),
+                                    assay = "RNA",
+                                    project = sample
```

Comment:
This is an example of where you would use a param. If you have a `sample_id` parameter defined, this would become:

```suggestion
                                    project = params$sample_id
```